distribution/lib/Standard/Base/0.0.0-dev/src/Data/Range.enso

```diff
@@ -302,14 +308,16 @@ Integer.up_to self n = case n of
 
    Arguments:
    - n: The end of the range.
+   - step: The step between consecutive elements of the range. It must be
+     non-zero. Defaults to 1. If positive it will be negated.
```

Comment:
If we are negating the step here, why not do the same (well the reverse) with `up_to`? It is inconsistent.

Still, I'm not convinced we should add `step` to `up_to`/`down_to` altogether. It does not read nice.

Although I do appreciate that the `10.down_to 0 . with_step 2` is not as nice in the GUI. But in general `up_to` and all these were not really made with the GUI in mind, in the first place.

What if instead of re-purposing these text-oriented methods we roll a separate one for usage in the GUI?

The current mix-up trying to work nicely both for text and GUI makes a mediocre experience for both IMHO.

Maybe we could introduce `Range.new start end step=1` or `Range.make` constructor that would be used in the GUI, instead of re-purposing these?

While I don't know what the widgets will look like, I imagine it can also play nicer - we could then have sliders both for start and end. Whereas with `x.up_to y` I don't think we'll be able to get a slider for the receiver `x`.